src/App.tsx

```diff
@@ -1,26 +1,215 @@
+import { StatusTodos } from './types/Status';
+
 
 const USER_ID = 0;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [status, setStatus] = useState<StatusTodos>(StatusTodos.ALL);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  const [isAddingTodo, setIsAddingTodo] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [updatingTodoId, setUpdatingTodoId] = useState<number | null>(null);
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [isDeletingTodo, setIsDeletingTodo] = useState<boolean>(false);
+
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setError(ErrorType.load_todo));
+  }, []);
+
+  const handleAddTodo = async (title: string, completed: boolean): Promise<Todo | undefined> => {
+    const trimmedTitle = title.trim()
+    if (!trimmedTitle) {
+      setError(ErrorType.empty_title);
+      return;
+    }
+
+    const newTodo: Omit<Todo, 'id'> = { title: trimmedTitle, userId: USER_ID, completed };
+
+
+
+    const tempTodoItem: Todo = { id: Date.now(), ...newTodo };
+    setTempTodo(tempTodoItem);
+    setIsAddingTodo(true);
+    setError(null);
+
+    try {
+      const createdTodo = await createTodos(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+      setTempTodo(null);
+
+      return createdTodo
+    } catch (error) {
+      setTempTodo(null);
+      setError(ErrorType.add_todo);
+    } finally {
+      setIsAddingTodo(false);
+    }
+  };
+
+  const handleUpdateTodoStatus = async (id: number, completed: boolean) => {
+    setUpdatingTodoId(id);
+
+    const todoToUpdate = todos.find(todo => todo.id === id);
+    if (!todoToUpdate) {
+      setError(ErrorType.not_found_todo);
+      return;
+    }
+
+    try {
+      const updatedTodo = await updateTodo({ ...todoToUpdate, completed });
+      setTodos(currentTodos =>
+        currentTodos.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+      );
+    } catch (e) {
+      setError(ErrorType.update_todo);
+    } finally {
+      setUpdatingTodoId(null);
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    setDeletingTodoId(todoId);
+    setIsDeletingTodo(true);
+    try {
+      await deleteTodo(todoId);
+
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch (e) {
+      setError(ErrorType.delete_todo);
+    } finally {
+      setDeletingTodoId(null);
+      setIsDeletingTodo(false);
+    }
+  };
+
+  const handleStatusChange = (newStatus: StatusTodos) => setStatus(newStatus);
+
+  const filteredTodos = todos.filter(todo => {
+    if (status === StatusTodos.ACTIVE) return !todo.completed;
+    if (status === StatusTodos.COMPLETED) return todo.completed;
+    return true;
```

Comment:
use switch case here and move filter inside switch to avoid extra iteration in default case